README.md

```diff
@@ -145,7 +172,7 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
+in javascript file title FSMStorey.js
```

Comment:
The file I have is called `index.js`.  If that's not the one you intended to send, let me know.

First: I appreciate you submitting a JavaScript program. :)

However, the program doesn't really capture the essence of state transitions. It's a perfectly good manual parser, but it doesn't use the table you'd previously created. I was looking for something like this:

~~~ js
const END = 99;

states = {
  0: { "the": 1 },
  1: { "lazy": 1, "smelly": 1, "dog": 2, "cat": 2 },
  2: { "ate": 3, "ran": 3 },
  3: { "slowly": 4, "noisily": 4, "EOI": END },
  4: { "EOI": END }
};

function getNextState(currentState, nextWord) {
  return states[currentState] && states[currentState][nextWord]
}

function stateMachine(sentence) {
  return sentence.reduce(getNextState, 0)
}
~~~

# -6